### DISCUSSION TITLE
Support for EFS throughput_mode set to elastic

### DISCUSSION THREAD

**Discussion by sewmiuraj** [gruntwork-io/knowledge-base#611] [Help] [answered]
Hello,

I need to confirm if gruntwork supports the value **elastic** for the variable throughput_mode.

The variables.tf says valid values are only: \"bursting\", \"provisioned\"
https://github.com/gruntwork-io/terraform-aws-data-storage/blob/main/modules/efs/variables.tf

If it is not supported, when can we expect the support to be enabled.

Many thanks.

**Comment by denis256** [comment]
hi,
from what I see `throughput_mode` is passed directly to `aws_efs_file_system` so it should accept `elastic` value

https://github.com/gruntwork-io/terraform-aws-data-storage/blob/main/modules/efs/main.tf#L39

**Comment by sewmiuraj** [comment]
Thank you for the confirmation.  

We have updated the reference to use the latest gruntwork release. But, it still won't let us use the throughput_mode as elastic. Could it be that when we run locally it pulls the hashicorp/aws v3.75.2 where it is not yet implemented?

![image](https://user-images.githubusercontent.com/96437456/207067050-71e3ae3a-d7ec-41e8-ab15-a42a4c17deae.png)

**Comment by denis256** [comment]
Hello,
indeed, I didn't check configured version - used [hashicorp/aws](https://github.com/gruntwork-io/terraform-aws-data-storage/blob/main/modules/efs/main.tf#L15) version 3.75.1 supports only `bursting` and `provisioned` https://registry.terraform.io/providers/hashicorp/aws/3.75.1/docs/resources/efs_file_system

Can be tried to upgrade to 4.x version but at this point, EFS module wasn't tested to work with 4.x

**Comment by james00012** [comment, marked as answer]
Hey @sewmiuraj, the latest release `v0.26.0` contain changes to upgrade the modules inside the data storage repo from 3.x.x to 4.22.0 which should give you the capability to use the feature you want. Please try it out and let us know if you have any further problems.
